GUID generation for dynamic fragments (#320)

Okeanos opened the pull request:
Add information on GUID generation based on [terminal#10374](https://github.com/microsoft/terminal/issues/10374) and improve wording to make the importance of a profile GUID clearer.

Additionally, I removed some trailing whitespace.

I believe that the section "How to determine the GUID of an existing profile" is still incomplete, though.

@DHowett provided instructions for how to generate a GUID for an existing dynamic fragment that was provided along with Git for Windows and presumably installed into `%ProgramData%\Microsoft\Windows Terminal\Fragments\Git\git-bash.json`:

> ```python
> import uuid
> 
> # CALCULATE NAMESPACE BASED ON FRAGMENT FOLDER NAME
> u = uuid.uuid5(uuid.UUID("{f65ddb7e-706b-4499-8a50-40313caf510a}"), "Git".encode("UTF-16LE").decode("ASCII"))
>
> # CALCULATE FINAL BASED ON NAMESPACE
> uuid.uuid5(u, "Git Bash".encode("UTF-16LE").decode("ASCII"))
> 
> # For Git/Git Bash: UUID('2ece5bfe-50ed-5f3a-ab87-5cd4baafed2b')
> ```

Note the UUID `f65ddb7e-706b-4499-8a50-40313caf510a` in there – I have no idea where this particular UUID comes from (the Windows Terminal namespace G/UUID is `2BDE4A90-D05F-401C-9492-E40884EAD1D8`). Using that one instead will result in a different GUID for `Git Bash` as profile name.

opbld30 commented:
Docs Build status updates of commit _[747c79f](https://github.com/Okeanos/terminal/commits/747c79f1a7311f5b46771e1e65eb17fe8a4622d4)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[TerminalDocs/json-fragment-extensions.md](https://github.com/Okeanos/terminal/blob/main/TerminalDocs/json-fragment-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/windows/terminal/json-fragment-extensions?branch=pr-en-us-460) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2021%5C12%5C12%5Cdd1c47df-1d46-d114-09b5-8ee205bcdb36%5CPullRequest%5C202112121221377419-460%5Cworkflow_report.html?sv=2020-08-04&se=2022-01-12T12%3A22%3A18Z&sr=b&sp=r&sig=XxIokX7hFr2ZID017Op6mN6DrZqWFQ%2BB0jmYDaiLJ2Q%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/dd1c47df-1d46-d114-09b5-8ee205bcdb36?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=master">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

cinnamon-msft commented on TerminalDocs/json-fragment-extensions.md:
```diff
@@ -59,31 +60,52 @@ The JSON file should be split up into 2 lists, one for profiles and one for sche
 
-- The namespace GUID: `{2BDE4A90-D05F-401C-9492-E40884EAD1D8}` 
-- The name of the profile to be updated 
+The GUIDs are generated using a Version 5 UUID generator which supports BOM-less UTF-16LE encoding. The namespace GUID for Windows Terminal is `{2BDE4A90-D05F-401C-9492-E40884EAD1D8}`.
```
```suggestion
The GUIDs are generated using a Version 5 UUID generator which supports BOM-less UTF-16LE encoding. The namespace GUID for Windows Terminal is `{f65ddb7e-706b-4499-8a50-40313caf510a}`.
```

Based on @DHowett's [comment](https://github.com/microsoft/terminal/issues/10374#issuecomment-857830194), I believe this is the correct namespace GUID.

cinnamon-msft commented on TerminalDocs/json-fragment-extensions.md:
```diff
@@ -59,31 +60,52 @@ The JSON file should be split up into 2 lists, one for profiles and one for sche
+import uuid
+
+# The Windows Terminal namespace GUID
+namespaceGUID = uuid.UUID("{2BDE4A90-D05F-401C-9492-E40884EAD1D8}")
```
```suggestion
namespaceGUID = uuid.UUID("{f65ddb7e-706b-4499-8a50-40313caf510a}")
```

Okeanos replied:
I updated the pull request accordingly.

DHowett commented:
Right, so that GUID comes from here:

https://github.com/microsoft/terminal/blob/a766798fb829a9e6f91706e919b71feabcb9b35e/src/cascadia/TerminalSettingsModel/Profile.h#L72

DHowett commented:
AH, and the guid `2BDE4A90-D05F-401C-9492-E40884EAD1D8` is for profiles that were generated by Windows Terminal itself. This is to disambiguate profiles we create (visual studio, powershell 7, ...) from profiles created by plugins or fragments so they can never accidentally collide. :smile:

opbld31 commented:
Docs Build status updates of commit _[526e8da](https://github.com/Okeanos/terminal/commits/526e8da4f19db615cf7ecc19ce05e6426ced35d0)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[TerminalDocs/json-fragment-extensions.md](https://github.com/Okeanos/terminal/blob/main/TerminalDocs/json-fragment-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/windows/terminal/json-fragment-extensions?branch=pr-en-us-460) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C1%5C7%5Cdd1c47df-1d46-d114-09b5-8ee205bcdb36%5CPullRequest%5C202201071325166665-460%5Cworkflow_report.html?sv=2020-08-04&se=2022-02-07T13%3A26%3A09Z&sr=b&sp=r&sig=QfO4vuKycXNwaWmm8Vn1sjoKdKCyPzr6QEahBbPSNR8%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/dd1c47df-1d46-d114-09b5-8ee205bcdb36?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

Okeanos commented:
@cinnamon-msft, @DHowett thank you for your feedback. I updated the pull request based on your comments and added a few additional sentences explaining the different namespace GUIDs. Please let me know if there's anything else you'd like changed.
Based on that feedback I also took the liberty of exchanging the "Ubuntu" GUID – I am wondering whether it'd be better to use ["Git Bash" example](https://github.com/microsoft/terminal/issues/10374#issuecomment-857830194) verbatim, though? The "Ubuntu" example is probably derived from the WSL-autogenerated profile?

DHowett commented:
> The "Ubuntu" example is probably derived from the WSL-autogenerated profile?

AH! You're totally right. Yes, using the Git Bash example would be perfect.

opbld31 commented:
Docs Build status updates of commit _[a0212a5](https://github.com/Okeanos/terminal/commits/a0212a536653363461990561580833075f0e959d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[TerminalDocs/json-fragment-extensions.md](https://github.com/Okeanos/terminal/blob/main/TerminalDocs/json-fragment-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/windows/terminal/json-fragment-extensions?branch=pr-en-us-460) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2022%5C1%5C7%5Cdd1c47df-1d46-d114-09b5-8ee205bcdb36%5CPullRequest%5C202201071438300722-460%5Cworkflow_report.html?sv=2020-08-04&se=2022-02-07T14%3A39%3A20Z&sr=b&sp=r&sig=SkPE7M9%2FYRvdDN9RLUr8FVYruaqlBYdInNnWwPJuRBs%3D).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/dd1c47df-1d46-d114-09b5-8ee205bcdb36?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

Okeanos commented:
> > The "Ubuntu" example is probably derived from the WSL-autogenerated profile?
> 
> AH! You're totally right. Yes, using the Git Bash example would be perfect.

I updated the pull request based on that. If there are any other changes you'd like me to make, please let me know.

mattwojo reviewed (approved):
Looks good!
